src/hotspot/share/gc/shenandoah/shenandoahBarrierSet.hpp

```diff
@@ -175,6 +175,8 @@ class ShenandoahBarrierSet: public BarrierSet {
     template <typename T>
     static oop oop_atomic_xchg_not_in_heap(T* addr, oop new_value);
 
+    template <typename T>
+    static void oop_store_common(T* addr, oop value);
```

Comment:
This is an implementation detail and a public part of the barrier set. We wouldn't expect users - other than the two existing callers - to ever use this api.

Can it be refactored out of this header and kept entirely within shenandoahBarrierSet.inline.hpp?

Reply:
I have made `oop_store_common` a private method. It needs to be part of `ShenandoahBarrierSet::AccessBarrier` to be able to access `RawAccessBarrier::oop_store()`.

Reply:
@DanHeidinga are you okay with the updates?

Reply:
Looks good to me